### PR TITLE
Aml 3774 add error to sign agreement page

### PR DESCRIPTION
**Pull request by MichaelYoung1981**
This is branched off AML-3754 so you'll need to change the base when doing the review.

### CODE REVIEW

File: src/SFA.DAS.EmployerAccounts.Web/Orchestrators/EmployerAgreementOrchestrator.cs [hunk 1]
```diff
@@ -87,7 +87,7 @@ protected EmployerAgreementOrchestrator()
         }
 
 
-        public async Task<OrchestratorResponse<EmployerAgreementViewModel>> GetById(
+        public virtual async Task<OrchestratorResponse<EmployerAgreementViewModel>> GetById(
```
Comment (Winsto) [inline src/SFA.DAS.EmployerAccounts.Web/Orchestrators/EmployerAgreementOrchestrator.cs]: Bit of a smell around having to make methods virtual just to test them, but I can see why that is and a refactor job to tidy up.

**Comment by Winsto** [comment]
Looks fine when I [compare with appropriate base](https://github.com/SkillsFundingAgency/das-employerapprenticeshipsservice/compare/AML-3574-legal-agreement...AML-3774_AddErrorToSignAgreementPage), but if I approve here the whole lot could get merged in. Do we : -
- do PRs back to a feature branch?
- wait for 3574 branch to go in first then approve these as smaller changes?

**Comment by MichaelYoung1981** [comment]
> Looks fine when I [compare with appropriate base](https://github.com/SkillsFundingAgency/das-employerapprenticeshipsservice/compare/AML-3574-legal-agreement...AML-3774_AddErrorToSignAgreementPage), but if I approve here the whole lot could get merged in. Do we : -
> 
> * do PRs back to a feature branch?
> * wait for 3574 branch to go in first then approve these as smaller changes?

Good question...in this instance the base is already in test so we can't merge into the base.  I'd say this can be approved (as the base already has been) but obviously shouldn't be tested until the other passes.

**Comment by Winsto** [comment]
> Looks fine when I compare with appropriate base, but if I approve here the whole lot could get merged in. Do we : -
> 
> do PRs back to a feature branch?
> wait for 3574 branch to go in first then approve these as smaller changes?
> 
> 
> Good question...in this instance the base is already in test so we can't merge into the base.  I'd say this can be approved (as the base already has been) but obviously shouldn't be tested until the other passes.

I will approve on the understanding it is not merged until the base branch is.

**Pull request merged**